flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointType.java

```diff
@@ -77,10 +93,31 @@ public String getName() {
         return name;
     }
 
+    public boolean canReuseFilesInSubsequentSnapshots() {
+        return sharingFilesStrategy == SharingFilesStrategy.FORWARD
+                || sharingFilesStrategy == SharingFilesStrategy.FORWARD_BACKWARD;
+    }
+
+    public boolean canReuseFilesFromPastSnapshots() {
+        return sharingFilesStrategy == SharingFilesStrategy.FORWARD_BACKWARD;
+    }
+
```

Comment:
I would be rather -1 for exposing this via boolean flags.

1. Combination of `!canReuseFilesInSubsequentSnapshots() && canReuseFilesFromPastSnapshots()` is not valid (at least at the moment)
2. It will make future compatibility more fragile. If we add another re-using mode exposed via another `canXYZ()` it will fail silently on state backends that do not support it. On the other hand exposing the enum directly `SharingFilesStrategy getSharingStrategy()` would fail (in correctly implemented users) with `throw new UnsupportedOperationException()`